packages/commonwealth/client/scripts/hooks/useWallets.tsx

```diff
@@ -0,0 +1,595 @@
+  const [cachedTimestamp, setCachedTimestamp] = useState<number>();
+  const [cachedChainId, setCachedChainId] = useState<string | number>();
+  const [primaryAccount, setPrimaryAccount] = useState<Account>();
+  const [secondaryLinkAccount, setSecondaryLinkAccount] = useState<Account>();
+  // const [secondaryChainId, setSecondaryChainId] = useState<string | number>();
+  const [isInCommunityPage, setIsInCommunityPage] = useState<boolean>();
+  const [isMagicLoading, setIsMagicLoading] = useState<boolean>();
+  const [showMobile, setShowMobile] = useState<boolean>();
+  const [signerAccount, setSignerAccount] = useState<Account>(null);
+  const [isNewlyCreated, setIsNewlyCreated] = useState<boolean>(false);
+  const [isLinkingOnMobile, setIsLinkingOnMobile] = useState<boolean>(false);
+
+  const isWalletConnectEnabled = _.some(
+    wallets,
+    (w) =>
+      (w instanceof WalletConnectWebWalletController ||
+        w instanceof TerraWalletConnectWebWalletController) &&
+      w.enabled
+  );
+
+  const isLinkingWallet = activeStep === 'selectPrevious';
+
+  useEffect(() => {
+    // Determine if in a community
+    const tempisInCommunityPage = app.activeChainId() !== undefined;
+    setIsInCommunityPage(tempisInCommunityPage);
+
+    if (tempisInCommunityPage) {
+      const chainbase = app.chain?.meta?.base;
+      setWallets(WebWalletController.Instance.availableWallets(chainbase));
+      setSidebarType('communityWalletOptions');
+      setActiveStep('walletList');
+    } else {
+      const allChains = app.config.chains.getAll();
+      const sortedChainBases = [
+        ChainBase.CosmosSDK,
+        ChainBase.Ethereum,
+        // ChainBase.NEAR,
+        ChainBase.Substrate,
+        ChainBase.Solana,
+      ].filter((base) => allChains.find((chain) => chain.base === base));
+      setWallets(
+        _.flatten(
+          sortedChainBases.map((base) => {
+            return WebWalletController.Instance.availableWallets(base);
+          })
+        )
+      );
+      setSidebarType('connectWallet');
+      setActiveStep('walletList');
+    }
+
+    setShowMobile(isWindowMediumSmallInclusive(window.innerWidth));
+
+    // Override if initial data is provided (needed for redirecting wallets + CommonBot)
+    if (walletProps.initialBody) {
+      setActiveStep(walletProps.initialBody);
+    }
+    if (walletProps.initialSidebar) {
+      setSidebarType(walletProps.initialSidebar);
+    }
+    if (walletProps.initialAccount) {
+      setPrimaryAccount(walletProps.initialAccount);
+      setAddress(walletProps.initialAccount.address);
+    }
+    // if (walletProps.initialWebWallet) {
+    //   this.selectedWallet = walletProps.initialWebWallet;
+    // }
+    if (walletProps.initialWallets) {
+      setWallets(walletProps.initialWallets);
+    }
+  }, []);
+
+  useEffect(() => {
+    // eslint-disable-next-line no-restricted-globals
+    addEventListener('resize', () =>
```

Comment:
can't we just use `window`?

Reply:
Nice spot. This was actually already there and I tried to stay close to existing implementation as much as possible.

Currently, I added a new hook https://github.com/hicommonwealth/commonwealth/pull/3714/commits/2ba8e538596d67a2fd2d31f509961a43de0b8abe to easily hook to window resize events. There are a lot of places where we can just dry up code with it

Example:
```
useBrowserWindow({
    onResize: () => { /* resize function */ } 
    resizeListenerUpdateDeps: [... deps that on change will update the resize listener],
  });
```